src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -316,6 +317,25 @@ public static void DetectUnsupportedModuleParameterAssignments(SemanticModel sem
             }
         }
 
+        public static void DetectInvalidValueForParentProperty(SemanticModel semanticModel, IDiagnosticWriter diagnosticWriter)
+        {
+            foreach (var resourceDeclarationSymbol in semanticModel.Root.ResourceDeclarations)
+            {
+                if (resourceDeclarationSymbol.TryGetBodyPropertyValue(LanguageConstants.ResourceParentPropertyName) is { } referenceParentSyntax)
+                {
+                    if (referenceParentSyntax is ArrayAccessSyntax arrayAccess)
+                    {
+                        referenceParentSyntax = arrayAccess.BaseExpression;
+                    }
+
+                    if (semanticModel.ResourceMetadata.TryLookup(referenceParentSyntax) is not { })
```

Comment:
It would be good to add a brief comment here explaining why we need to block this case